src/components/DatePicker/DatePicker.tsx

```diff
@@ -10,14 +10,42 @@ import { DatePickerTrigger } from './DatePickerTrigger';
 import { IconCaretLeft, IconCaretLeftDouble, IconCaretRight, IconCaretRightDouble } from '@foundation/Icon';
 import { Validation } from '@utilities/validation';
 
+type SingleDatePickerProps = {
+    variant?: 'single';
+    onChange: (date: Date | null) => void;
+    selectsRange?: false;
+    startDate?: null;
+    endDate?: null;
+};
+
+type RangeDatePickerProps = {
+    variant: 'range';
+    onChange: (date: [Date | null, Date | null] | null) => void;
+    selectsRange: true;
+    startDate: Date | null;
+    endDate: Date | null;
+};
+
 export type DatePickerProps = {
     placeHolder?: string;
     isClearable?: boolean;
     shouldCloseOnSelect?: boolean;
-    onChange?: (date: Date | null) => void;
     dateFormat?: string;
-    value?: Date;
+    value?: Date | null;
+    minDate?: Date;
+    maxDate?: Date;
     validation?: Validation;
+    customTrigger?: React.ReactNode;
+    children?: React.ReactNode;
+    hasPopperArrow?: boolean;
+    filterDate?: (date: Date) => boolean;
```

Comment:
What's filterDate for?

Reply:
Allows further narrowing of the selectable dates: https://reactdatepicker.com/#example-filter-dates